src/main/java/com/kosa/mini/api/service/member/MemberEditProfileServiceImpl.java

```diff
@@ -2,62 +2,96 @@
         this.passwordEncoder = passwordEncoder;
         this.storeRepository = storeRepository;
     }
 
+    // 사용자 정보 조회
+    public MemberEditProfileDTO getUserInfo(Integer memberId) {
+        // 1. 사용자를 조회
+        Member member = memberRepository.findById(memberId).orElse(null);
+        if (member == null) {
+            return null; // 사용자가 없으면 null 반환
+        }
+
+        // 2. Member 정보를 DTO로 변환
+        MemberEditProfileDTO memberDTO = new MemberEditProfileDTO();
+        memberDTO.setMemberId(member.getMemberId());
+        memberDTO.setName(member.getName());
+        memberDTO.setEmail(member.getEmail());
+        memberDTO.setNickname(member.getNickname());
+        memberDTO.setPhoneNumber(member.getPhoneNumber());
+        memberDTO.setRoleId(member.getRole());
+
+        // 3. Role이 사장님(3)인 경우 가게 이름 추가
+        if (member.getRole().getRoleId() == 3) { // 사장님
+            Store store = storeRepository.findByOwner_MemberId(memberId);
+            if (store != null) {
+                memberDTO.setStoreName(store.getStoreName());
+            }
+        }
+
+        return memberDTO;
+    }
+
+    // 사용자 정보 수정
     public Member update(Integer memberId, MemberEditProfileDTO dto) {
         dto.setPassword(passwordEncoder.encode(dto.getPassword()));
-        log.info("암호화  pw"+ dto.getPassword());
+        log.info("암호화  pw: " + dto.getPassword());
         Member member = dto.toEntity();
-        log.info("password 인코딩"+member.getPassword());
-        log.info("id: "+memberId.toString()+"member: "+member.toString());
-        // 2. id 조회하기
+        log.info("password 인코딩: " + member.getPassword());
+        log.info("id: " + memberId + " member: " + member);
+
+        // 1. id 조회
         Member members = memberRepository.findById(memberId).orElse(null);
-        // 3. 예외 처리
-        if(members == null || memberId != member.getMemberId()) {
-            //400, 잘못된 요청 응답!
-            log.info("잘못된 요청! id : "+memberId.toString()+"member: "+member.toString());
+        if (members == null || !memberId.equals(member.getMemberId())) {
+            log.info("잘못된 요청! id : " + memberId + " member: " + member);
             return null;
         }
-        //비밀번호 암호화
-        if(dto.getPassword() != null && !dto.getPassword().isEmpty()) {
+
+        // 2. 비밀번호 수정
+        if (dto.getPassword() != null && !dto.getPassword().isEmpty()) {
             members.setPassword(dto.getPassword());
         }
 
-        //4. 업데이트 및 정상 응답코드(200)날리기
-        members.put(member);
-        Member updated = memberRepository.save(members);
-        return updated;
-    }
+        // 3. Role이 사장님(3)인 경우 가게 정보 업데이트
+        if (member.getRole().getRoleId() == 3) { // 사장님 Role ID가 3
+            Store store = storeRepository.findByOwner_MemberId(memberId);
+            if (store != null) {
+                store.setStoreName(dto.getStoreName());
+                storeRepository.save(store);
+            }
+        }
```

Comment:
```suggestion
        // 3. Role이 사장님(3)인 경우 가게 정보 업데이트
        if (member.getRole().getRoleId() == 3) { // 사장님 Role ID가 3
            Store store = storeRepository.findByOwner_MemberId(memberId);
            if (store != null) {
                store.setStoreName(dto.getStoreName());
                storeRepository.save(store);
            }
        }
```

가게 수정 되면 안됩니다~ 관리자만 가능해요 ^^
조건 잘 확인하셔야해요~